test/test-harness.sh

```diff
@@ -36,12 +36,15 @@ overall_status=0
 function run(){
     local temp_dir=$1
     shift
+    # This variable is used to override the default value in the kubelet mock
+    KUBELET_VERSION="${KUBELET_VERSION:-}"
     cp -f ${SCRIPTPATH}/../files/kubelet-config.json ${temp_dir}/kubelet-config.json
     docker run -v ${SCRIPTPATH}/../files/:/etc/eks/ \
         -v ${temp_dir}/kubelet-config.json:/etc/kubernetes/kubelet/kubelet-config.json \
         --attach STDOUT \
         --attach STDERR \
         --rm \
+        -e KUBELET_VERSION="$KUBELET_VERSION" \
         eks-optimized-ami $@
```

Comment:
We need to tweak the mocking strategy to allow individual test cases to supply their own mocks. Not sure we should address it in this PR, just a call out

Reply:
Ya, that's fair. Do you hate overwriting `kubelet` as part of the test as an alternative? That might be more sensible, but seems a little wrong in its own way.

Reply:
I didn't actually address this. Definitely something to consider down the road.